.github/autobuild/ios.sh

```diff
@@ -30,17 +30,76 @@ setup() {
     fi
 }
 
+
+prepare_signing() {
+    [[ "${SIGN_IF_POSSIBLE:-0}" == "1" ]] || return 1
+
+    # Signing was requested, now check all prerequisites:
+    [[ -n "${IOSDIST_CERTIFICATE:-}" ]] || return 1
+    [[ -n "${IOSDIST_CERTIFICATE_ID:-}" ]] || return 1
+    [[ -n "${IOSDIST_CERTIFICATE_PWD:-}" ]] || return 1
+    [[ -n "${NOTARIZATION_PASSWORD:-}" ]] || return 1
+    [[ -n "${IOS_PROV_PROFILE_B64:-}" ]] || return 1
+    [[ -n "${KEYCHAIN_PASSWORD:-}" ]] || return 1
+
+    echo "Signing was requested and all dependencies are satisfied"
+
+    # use this as filename for Provisioning Profile
+    IOS_PP_PATH="embedded.mobileprovision"
+
+    ## Put the cert to a file
+    # IOSDIST_CERTIFICATE - iOS Distribution
+    echo "${IOSDIST_CERTIFICATE}" | base64 --decode > iosdist_certificate.p12
+
+    ## Echo Provisioning Profile to file
+    echo -n "${IOS_PROV_PROFILE_B64}" | base64 --decode > $IOS_PP_PATH
+
+    # Set up a keychain for the build:
+    security create-keychain -p "${KEYCHAIN_PASSWORD}" build.keychain
+    security default-keychain -s build.keychain
+    security unlock-keychain -p "${KEYCHAIN_PASSWORD}" build.keychain
+    security import iosdist_certificate.p12 -k build.keychain -P "${IOSDIST_CERTIFICATE_PWD}" -A -T /usr/bin/codesign
+    # security set-key-partition-list -S apple-tool:,apple:,codesign: -s -k "${KEYCHAIN_PASSWORD}" build.keychain
+    security set-key-partition-list -S apple-tool:,apple: -s -k "${KEYCHAIN_PASSWORD}" build.keychain
+    # set lock timeout on keychain to 6 hours
+    security set-keychain-settings -lut 21600
+    
+    # apply provisioning profile
+    #FIXME - maybe redundant?
+    mkdir -p ~/Library/MobileDevice/Provisioning\ Profiles
+    cp $IOS_PP_PATH ~/Library/MobileDevice/Provisioning\ Profiles
+
+    # for debug
+    echo "Checking found identities..."
+    security find-identity -v 
+
+    # Tell Github Workflow that we need notarization & stapling:
+    echo "::set-output name=macos_signed::true"
+    return 0
+}
+
 build_app_as_ipa() {
     # Add the Qt binaries to the PATH:
     export PATH="${QT_DIR}/${QT_VERSION}/ios/bin:${PATH}"
-    ./ios/deploy_ios.sh
+
+    # Mac's bash version considers BUILD_ARGS unset without at least one entry:
+    BUILD_ARGS=("")
+    if prepare_signing; then
+        BUILD_ARGS=("-s" "${IOSDIST_CERTIFICATE_ID}" "-k" "${KEYCHAIN_PASSWORD}")
+    fi
+    ./ios/deploy_ios.sh "${BUILD_ARGS[@]}"
 }
 
 pass_artifact_to_job() {
-    local artifact="jamulus_${JAMULUS_BUILD_VERSION}_iOSUnsigned${ARTIFACT_SUFFIX:-}.ipa"
+    local artifact="jamulus_${JAMULUS_BUILD_VERSION}_iOS_unsigned${ARTIFACT_SUFFIX:-}.ipa"
```

Comment:
Not sure how much a signed IPA on the web without the app store helps?

Reply:
I don't think there's any point to that, no - the signed IPA is purely for App Store submission. So it's really for @emlynmac or whoever is managing the Jamulus App Store account to have a ready-built file for submission.

I assume the unsigned IPA is useful for general testing purposes so obviously I've kept that in.